### PR TITLE
chore: clean up abstractions used in CLI command

### PR DESCRIPTION
**Pull request by kevinslin**
CLI returns a `RespV3` object which is either `data: T` or `DendronError`. 
Use a typescript union to return a type that is either data or error. This results in a stanrad type safe way of checking for errors from any response. 

[Docs](https://github.com/dendronhq/dendron-site/commit/36406061c26b66525497f1ee4f9090b89f0c5b17)

### CODE REVIEW

File: packages/pods-core/src/builtin/AirtablePod.ts [hunk 1]
```diff
@@ -101,15 +101,15 @@ export class AirtableUtils {
...
+    // let total: number = 0;
     const out = await Promise.all(
       chunks.flatMap(async (record) => {
         // @ts-ignore
         await limiter.removeTokens(1);
         const data = JSON.stringify({ records: record });
         try {
           const _records = await func(record);
-          total += _records.length;
+          // total += _records.length;
```
Comment (jonathanyeung) [inline packages/pods-core/src/builtin/AirtablePod.ts:112]: should we just remove these?

**Review by jonathanyeung — CHANGES_REQUESTED** [review]

File: packages/pods-core/src/builtin/AirtablePod.ts (hunk 1 above)
Comment (kevinslin) [reply packages/pods-core/src/builtin/AirtablePod.ts:112]: yep, done

File: packages/common-all/src/error.ts [hunk 2]
```diff
@@ -217,6 +217,16 @@ export function assertInvalidState(msg: string): never {
 
 /** Utility class for helping to correctly construct common errors. */
 export class ErrorFactory {
+  /**
+   * Not found
+   */
+  static create404Error({ url }: { url: string }): DendronError {
+    return new DendronError({
+      message: `resource at ${url} did not exist`,
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply packages/common-all/src/error.ts:225]: fixed

File: packages/common-all/src/types/typesv2.ts [hunk 3]
```diff
@@ -190,6 +190,19 @@ export interface RespV2<T> {
...
+ * This lets us use a discriminate union to see if result has error or data
+ */
+export type RespV3<T> =
+  | {
+      error: IDendronError;
+      data?: never;
+    }
+  | {
+      error?: never;
+      data: T;
```
[the remark this answers is not in the document]
Comment (kevinslin) [reply packages/common-all/src/types/typesv2.ts:203]: It's a fair point. This return type doesn't allow it. For most of the cases where we return an error, we've been doing the "error or data" approach. 

The current solution gets the typescript compiler to do the typecheck in this common case.

If we were to introduce the `error?, data?` approach, the issue is that we would go back to having to check for the error and having to check for the data (or applying a type assertion).

the tradeoff with this solution is developer ergonomics vs being able to express a minor error. 

My proposal here is to keep this but introduce a new error for the case where minor errors are allowed.

```ts
// current implementation
type RespEitherDataOrError

// more general implementation
type RespDataOrError
```

A developer can use the latter if minor errors are something that their program returns.
Thoughts?

Open to changing my mind on this

**Comment by kevinslin** [comment]
pr comments addressed in this new PR -> https://github.com/dendronhq/dendron/pull/1716

**Pull request closed without merging**